pkg/oc/cli/cmd/newapp.go

```diff
@@ -148,6 +149,33 @@ type NewAppOptions struct {
 	*ObjectGeneratorOptions
 }
 
+func checkResources(discoveryClient discovery.DiscoveryInterface) error {
+	resources, err := discoveryClient.ServerResources()
+	if err != nil {
+		return fmt.Errorf("Unable to to get list of available resources: %v", err)
+	}
+
+	resourceMap := make(map[string]bool)
+	for _, resourceList := range resources {
+		for _, resource := range resourceList.APIResources {
+			resourceMap[resource.Kind] = true
+		}
+	}
+
+	requiredResources := []string{"BuildConfig", "ImageStream"}
```

Comment:
we actually have both the legacy apis and the grouped apis.  This should probably check to see if either exists, in case we eventually remove the legacy api.

Reply:
I don't think you have to check for the ImageStream resource as there is no way today to disable image apis. So this can be simplified to just check the existence of BuildConfigs. 

Also I don't think we support legacy (oapi) in CLI anymore, right @deads2k / @soltysh  ?

Reply:
> Also I don't think we support legacy (oapi) in CLI anymore, right @deads2k / @soltysh ?

No legacy in the CLI.

Reply:
Thanks, I will mark the PR as [WIP] for now and address the comments when I'm back from PTO next week.

Reply:
What does "no legacy in the cli" mean?  In this case we're simply checking what the api server supports, we're not using these apis to call the server.